### PR TITLE
chore: use vitest, improve ci

### PR DESCRIPTION
**Pull request by Uzlopak**
use vitest, break the ci in separate jobs

### CODE REVIEW

**Comment by wolfy1339** [comment]
I think it might be a good idea to update the demo tests to a newer version of deps

**Comment by Uzlopak** [comment]
done ;)

**Review by wolfy1339 — DISMISSED** [review]

**Review by wolfy1339 — DISMISSED** [review]

File: test/deno/web_test.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
 import { sign, verify } from "../../pkg/dist-web/index.js";
```
Comment (wolfy1339) [inline test/deno/web_test.ts:1]: It seems this import is out of date now.

Comment (Uzlopak) [reply test/deno/web_test.ts:1]: for browser we need to build it once

Comment (wolfy1339) [reply test/deno/web_test.ts:1]: The `dist-web` folder doesn't exist anymore. It should be `dist-bundle`

**Review by wolfy1339 — DISMISSED** [review]

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**